clients/rospy/src/rospy/core.py

```diff
@@ -143,78 +145,107 @@ def rospywarn(msg, *args):
     """Internal rospy client library warn logging"""
     _rospy_logger.warn(msg, *args)
     
-logdebug = logging.getLogger('rosout').debug
 
-logwarn = logging.getLogger('rosout').warning
+def _base_logger(msg, *args, **kwargs):
+
+    try:
+        name = kwargs.pop('logger_name')
+    except KeyError:
+        name = None
+
+    try:
+        throttle = kwargs.pop('logger_throttle')
+    except KeyError:
+        throttle = None
+
+    try:
+        level = kwargs.pop('logger_level')
+    except KeyError:
+        level = None

```

Comment:
I had the function using default values in one iteration but I believe i started failing tests and decided to use the keyword arguments instead.  I'll rerun the tests and see if I can remember why I switched to using keyword args instead.